mydumper.c

```diff
@@ -1998,9 +2024,13 @@ void dump_create_database_data(MYSQL *conn, char *database, char *filename) {
 
   if (!compress_output)
     outfile = g_fopen(filename, "w");
-  else
-    outfile = (void *)gzopen(filename, "w");
-
+  else {
+      char *command = g_strdup_printf("%s > %s", compress_command,filename);
+        outfile = popen(command, "w");
```

Comment:
The `popen` call can be very dangerous. Is input sanitization warranted, here, or at least a *big scary warning* about delegating to the shell?